libs/api/domains/auth-admin/src/lib/application/applications.service.ts

```diff
@@ -23,27 +24,52 @@ export class ApplicationsService {
     return resp
   }
 
-  createApplication(input: CreateApplicationInput) {
-    const displayName = new TranslatedValue()
-    displayName.locale = 'is'
-    displayName.value = input.displayName
+  async createApplication(
+    input: CreateClientsInput,
+    user: Auth,
+  ): Promise<CreateApplicationResponseDto[]> {
+    const applicationRequest = {
+      tenantId: input.tenantId,
+      adminCreateClientDto: {
+        clientId: input.applicationId,
+        clientType: (input.applicationType as string) as AdminCreateClientDtoClientTypeEnum,
+        tenantId: input.tenantId,
+        nationalId: user.nationalId,
+        clientName: input.displayName,
+      },
+    }
 
-    const application = new Application()
-    application.applicationId = input.applicationId
-    application.applicationType = input.applicationType
-    application.environments = input.environments.map((env) => {
-      const applicationEnv = new ApplicationEnvironment()
-      applicationEnv.name = input.applicationId
-      applicationEnv.environment = env
-      applicationEnv.displayName = [displayName]
-      applicationEnv.ApplicationUrls.callbackUrls = []
+    const createApplicationResponse = [] as CreateApplicationResponseDto[]
 
-      return applicationEnv
-    })
+    const created = await Promise.all(
+      input.environments.map(async (environment) => {
+        switch (environment) {
+          case Environment.Development:
+            return this.adminDevApiWithAuth(user)?.meClientsControllerCreate(
+              applicationRequest,
+            )
+          case Environment.Staging:
+            return this.adminStagingApiWithAuth(
+              user,
+            )?.meClientsControllerCreate(applicationRequest)
+          case Environment.Production:
+            return this.adminProdApiWithAuth(user)?.meClientsControllerCreate(
+              applicationRequest,
+            )
+        }
```

Comment:
We should move this to the MultiEnvironment abstract class as `adminApiByEnvironmentWithAuth` helper so we can use it in other methods as update and delete.

```
adminApiByEnvironmentWithAuth(environment: Environment, user: User) {
       switch (environment) {
          case Environment.Development:
            return this.adminDevApiWithAuth(user)?
          case Environment.Staging:
            return this.adminStagingApiWithAuth(user)?
          case Environment.Production:
            return this.adminProdApiWithAuth(user)?
        }
}
```


```suggestion
        return this.adminApiByEnvironmentWithAuth(environment, user)?.meClientsControllerCreate(
          applicationRequest,
        )
```